### PR TITLE
Fix for #96, Italics parsing broken

### PR DESCRIPTION
**Pull request by anderskaplan**
The issue was that smart quotes, as well as any other non-ascii punctuation characters, were not handled like ascii punctuation in the parsing of emphasis/strong tokens.
Solved by including all unicode punctuation in the set of punctuation characters.

### CODE REVIEW

**Review by pbodnar — COMMENTED** [review]
@anderskaplan, thanks for another nice fix!

It took me some time and research to confirm, but it looks OK and it also doesn't seem to affect performance (`$ py test/benchmark.py mistletoe`). So just 1 or 2 minor improvement suggestions from me...

File: mistletoe/core_tokens.py [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
 whitespace = {' ', '\t', '\n', '\x0b', '\x0c', '\r'}
 unicode_whitespace = {'\t', '\n', '\x0b', '\x0c', '\r', '\x1c', '\x1d', '\x1e',
         '\x1f', ' ', '\x85', '\xa0', '\u1680', '\u2000', '\u2001', '\u2002',
         '\u2003', '\u2004', '\u2005', '\u2006', '\u2007', '\u2008', '\u2009',
         '\u200a', '\u2028', '\u2029', '\u202f', '\u205f', '\u3000'}
-punctuation = {'!', '"', '#', '$', '%', '&', '\'', '(', ')', '*', '+', ',',
-               '-', '.', '/', ':', ';', '<', '=', '>', '?', '@', '[', '\\',
-               ']', '^', '_', '`', '{', '|', '}', '~'}
+unicode_chrs = (chr(i) for i in range(sys.maxunicode + 1))
+punctuation = set.union(
```
[the remark this answers is not in the document]
Comment (anderskaplan) [reply mistletoe/core_tokens.py:12]: Ok, sure, I'll add that comment.

I was also on the same track as you at first, thinking that the ascii punctuation should be a subset of the unicode, but there are some characters in there, like the dollar sign, which are classified differently in unicode.

File: test/test_span_token.py [hunk 2]
```diff
@@ -35,6 +35,15 @@ def test_parse(self):
         self._test_parse(span_token.Emphasis, '*some text*', 'some text')
         self._test_parse(span_token.Emphasis, '_some text_', 'some text')
 
+    def test_emphasis_with_straight_quote(self):
```
[the remark this answers is not in the document]
Comment (anderskaplan) [reply test/test_span_token.py:38]: I just checked: none of the failing tests in the latest CommonMark test set are caused by this issue. So no, these cases aren't covered by that test set. I agree that's a bit surprising.

File: mistletoe/core_tokens.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pbodnar) [reply mistletoe/core_tokens.py:16]: Thanks.

File: test/test_span_token.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pbodnar) [reply test/test_span_token.py:38]: OK, thanks for adding the tests.

**Pull request merged**